components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/IdentityProviderDataRetrievalClient.java

```diff
@@ -102,6 +111,127 @@ public String getIdPImage(String tenant, String idpName)
+        if (federatedAuthenticators == null) {
+            return configMap;
+        }
+        String federatedIDPURLWithID = null;
+        for (int i = 0; i < federatedAuthenticators.length(); i++) {
+            JSONObject property = (JSONObject) federatedAuthenticators.get(i);
+            if (idpCode.equals(property.getString(NAME))) {
+                federatedIDPURLWithID = property.getString(SELF_LINK);
+                break;
+            }
+        }
+        if (StringUtils.isEmpty(federatedIDPURLWithID)) {
+            return configMap;
+        }
+
+        // i.e. /t/carbon.super/api/server/v1/identity-providers/6719c5cc-5162-44c7-9190-cb74d500f5fc/
+        //               federated-authenticators/R29vZ2xlT0lEQ0F1dGhlbnRpY2F0b3I
+        JSONObject federatedIdpResult = executePath(tenant, federatedIDPURLWithID);
+        if (federatedIdpResult == null) {
+            return configMap;
+        }
+        JSONArray federatedIdpProperties = federatedIdpResult.getJSONArray(PROPERTIES);
+        if (federatedIdpProperties.length() == 0) {
+            return configMap;
+        }
+        for (int i = 0; i < federatedIdpProperties.length(); i++) {
+            JSONObject property = (JSONObject) federatedIdpProperties.get(i);
+            if (configKeys.contains(property.getString(KEY))) {
+                configMap.putIfAbsent(property.getString(KEY), property.getString(VALUE));
+            }
+        }
+        return configMap;
+    }
+
+    /**
+     * This function executes the HTTP GET calls of given path on carbon API and returns the resulted JSON object
+     *
+     * @param tenant String. The tenant name of the session i.e. carbon.super.
+     * @param path   String. The context or to call without host details. i.e.
+     *               api/server/v1/identity-providers?filter=name+eq+Google
+     *               api/server/v1/identity-providers/6719c5cc-5162-44c7-9190-cb74d500f5fc
+     * @return JSONObject. The resulted JSONObject of the GET call
+     * @throws IdentityProviderDataRetrievalClientException When there is an error with executing the get calls
+     */
+    private JSONObject executePath(String tenant, String path) throws IdentityProviderDataRetrievalClientException {
+
+        JSONObject jsonResponse = null;
+        try (CloseableHttpClient httpclient = HttpClientBuilder.create().useSystemProperties().build()) {
+            String url = getEndpoint(tenant, path);
+            HttpGet httpGet = new HttpGet(url);
+            setAuthorizationHeader(httpGet);
+
+            try (CloseableHttpResponse response = httpclient.execute(httpGet)) {
+                if (response.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
+                    jsonResponse = new JSONObject(
+                            new JSONTokener(new InputStreamReader(response.getEntity().getContent())));
+                }
+            } finally {
+                httpGet.releaseConnection();
```

Comment:
```suggestion
                EntityUtils.consume(response.getEntity());
```

Reply:
Since try-with block release all the resources, I think this is redundant. WDYT ? (Returned exceptions after adding this)